### DISCUSSION TITLE
page.get_text('words', myRect) versus page.search_for()

### DISCUSSION THREAD

**Discussion by sgthuth** [pymupdf/PyMuPDF#2860] [Q&A] [answered]
Hopefully, this is an easy one.

I have a PDF document and on particular pages is a string of characters.  If myString is there, I have to perform various actions.

Out of the 18 page document, I can easily see there are 10 pages that contain myString.  Trying to automate additional steps, I used page.search_for(myString).  

Performing page.search_for(myString) only results in 6 pages where myString is found; not 10.

As a means of troubleshooting, I used page.get_text('words', pageDimension) and printed them to the console.  I can see myString within the list of words on all 10 pages.

Is there anything off hand that could prevent the search from picking up myString?

**Comment by JorjMcKie** [comment]
maybe the needle contains accented characters? In that case the casing is important, e.g. "enchantÉ" will not be found when searching for "enchanté".
Otherwise - am a bit uncertain, have a problem page to look at?

**Reply by sgthuth** [reply]
I don't remember all the pages search_for() did not find myString but I know on page 3, it did not as expected.

**Reply by JorjMcKie** [reply, marked as answer]
Looks ok to me:
```python
In [1]: import fitz
In [2]: doc=fitz.open("T3683153_Preped.PDF")
In [3]: page=doc[0]
In [4]: page=doc[2]
In [5]: page.search_for("dps$$$PKG")
Out[5]: [Rect(72.0, 150.3087615966797, 130.6758575439453, 162.4904022216797)]
In [6]: [w for w in page.get_text("words") if w[4]=="DPS$$$PKG"]
Out[6]:
[(72.0,
  150.3087615966797,
  130.6758575439453,
  162.4904022216797,
  'DPS$$$PKG',
  2,
  0,
  0)]
```

**Reply by sgthuth** [reply]
I'll have to do a little more digging than.  Sorry to have bothered you.

**Reply by JorjMcKie** [reply]
np - may be the wrong example

**Comment by JorjMcKie** [comment]
Of course: if the needles are not fully contained in page.rect, then there also can occur problems.
To be absolutely sure use `clip=fitz.INFINITE_RECT()` on all extractions / searches. This will prevent omitting things only partially intersecting the page rectangle.

**Reply by sgthuth** [reply]
Is the clip used in the search_for() method as a parameter?

Like...
search_for(myString, clip)

**Reply by JorjMcKie** [reply]
yes, clip is an omnipresent parameter: all get_text() variants (but not in HTML, XML, XHTML), search_for() and get_pixmap().
Similar text extraction flags.
